### ISSUE TITLE
Cloudflare is blocked in Indonesia (so the apt repo at termux.net does not work there)

### ISSUE THREAD

**Issue by kobor03** [termux-play-store/termux-issues#30]
### Problem description

$ apt update
Ign:1 https://termux.net stable InRelease
Ign:1 https://termux.net stable InRelease
Ign:1 https://termux.net stable InRelease
Err:1 https://termux.net stable InRelease
  Certificate verification failed: The certificate is NOT trusted. The certificate issuer is unknown. The name in the certificate does not match the expected.  Could not handshake: Error in the certificate verification. [IP: 36.86.63.185 443]
All packages are up to date.
Warning: Failed to fetch https://termux.net/dists/stable/InRelease  Certificate verification failed: The certificate is NOT trusted. The certificate issuer is unknown. The name in the certificate does not match the expected.  Could not handshake: Error in the certificate verification. [IP: 36.86.63.185 443]
Warning: Some index files failed to download. They have been ignored, or old ones used instead.
~ $ termux-change-repo
termux-change-repo: command not found

### What steps will reproduce the bug?

$ apt update
Ign:1 https://termux.net stable InRelease
Ign:1 https://termux.net stable InRelease
Ign:1 https://termux.net stable InRelease
Err:1 https://termux.net stable InRelease
  Certificate verification failed: The certificate is NOT trusted. The certificate issuer is unknown. The name in the certificate does not match the expected.  Could not handshake: Error in the certificate verification. [IP: 36.86.63.185 443]
All packages are up to date.
Warning: Failed to fetch https://termux.net/dists/stable/InRelease  Certificate verification failed: The certificate is NOT trusted. The certificate issuer is unknown. The name in the certificate does not match the expected.  Could not handshake: Error in the certificate verification. [IP: 36.86.63.185 443]
Warning: Some index files failed to download. They have been ignored, or old ones used instead.
~ $ termux-change-repo
termux-change-repo: command not found

### System information

```shell
$ apt update
Ign:1 https://termux.net stable InRelease
Ign:1 https://termux.net stable InRelease
Ign:1 https://termux.net stable InRelease
Err:1 https://termux.net stable InRelease
  Certificate verification failed: The certificate is NOT trusted. The certificate issuer is unknown. The name in the certificate does not match the expected.  Could not handshake: Error in the certificate verification. [IP: 36.86.63.185 443]
All packages are up to date.
Warning: Failed to fetch https://termux.net/dists/stable/InRelease  Certificate verification failed: The certificate is NOT trusted. The certificate issuer is unknown. The name in the certificate does not match the expected.  Could not handshake: Error in the certificate verification. [IP: 36.86.63.185 443]
Warning: Some index files failed to download. They have been ignored, or old ones used instead.
~ $ termux-change-repo
termux-change-repo: command not found
```

**Comment by fornwall** [comment]
Thanks for reporting!

> Warning: Failed to fetch https://termux.net/dists/stable/InRelease Certificate verification failed: The certificate is NOT trusted. The certificate issuer is unknown. The name in the certificate does not match the expected. Could not handshake: Error in the certificate verification. [IP: 36.86.63.185 443]

Could you try the following:
1. What is the output of running `curl -iv -o /dev/null https://termux.net` - could you paste the output?
2. Does it work to connect to https://termux.net/dists/stable/InRelease in your normal Android browser?
3. Where are you located (in which country)?
4. Could you paste the output from running `termux-info`?

> ~ $ termux-change-repo
> termux-change-repo: command not found

The `termux-change-repo` command is not yet available, as the Google Play build currently uses a separate package repository that is not mirrored. Will most likely be fixed in the future by switching to the main package repository.

**Comment by nhockey** [comment]
It looks to me like something is trying to cache your https request (MiTM). That IP is someplace in Indonesia. 

The addresses I got for the actual website;
IPv6: 2606:4700:3032::6815:217b, 2606:4700:3036::ac43:a28a
IPv4: 104.21.33.123, 172.67.162.138
Which looks like a cloudflare CDN; but the ip address that you listed definitely is NOT part of the cloudflare CDN. 

I would recommend going in and setting a private DNS server in your Android settings and seeing if that resolves the problem.

You also might want to consider going into your device's key store and making sure that there isn't a custom certificate authority that's verifying certificates that aren't really authentic.

**Comment by RadiatedExodus** [comment]
> It looks to me like something is trying to cache your https request (MiTM). That IP is someplace in Indonesia.

Adding onto this, this is (most likely) the ip that is used by the government to show blocked websites. The ip is indeed in Jakarta, Indonesia, and it's owned by ``PT. Telkom Indonesia``. Most likely op is using Indihome. If you access the ip with ``https`` (``https://36.86.63.185``) it should show the block web page.

Tl;dr; domain is blocked by Indonesian government

A temporary solution, is to use another dns resolver like Cloudflare's 1.1.1.1, if that doesn't work then ensure the isp isn't also spoofing/blocking dns resolvers (I'm confident that Telkom does, I've tried before with 1.1.1.1, it would head to their dns resolver instead of Cloudflare's)

If the solution above doesn't work, use a VPN, or change your isp to another one that is a bit more lenient on the domain blocking (NOTE: Take with a grain of salt, I could never confirm the fact that if all isps in Indonesia has switched to the government blocklist or still maintain their own based on the government's requirements)

**Comment by fornwall** [comment]
The plan is to switch to the same package repository that is used by the F-Droid build of Termux which has multiple mirrors: https://github.com/termux/termux-packages/wiki/Mirrors (currently the Google Play build uses a custom package repository, as all necessary changes are not yet merged).

Until that is done, one can install Termux from https://f-droid.org/en/packages/com.termux/ to be able to switch to package mirrors using the `termux-change-repo` command (note that one needs to uninstall the app before installing it again from a different source).

**Comment by nhockey** [comment]
i don't think the government would block the entirety of cloudflare's CDN; that would break 90% of the internet. It has to be a DNS interception. :thinking: 
I would try to enable DNS over TLS so your lookups are encrypted. If you are running a version of android newer then 9 or 10 you can enable that feature this way usually:
1. Go to Settings > Network & internet.
2. Select Advanced > Private DNS.
3. Select the Private DNS provider hostname option.
4. Enter one.one.one.one and select Save.

**Comment by RadiatedExodus** [comment]
> i don't think the government would block the entirety of cloudflare's CDN; that would break 90% of the internet. It has to be a DNS interception. 🤔

Usually, isps in Indonesia implement the blocking in the form of dns poisoning (which most, if not all, still does do it in the form of dns poisoning)

**Comment by fornwall** [comment]
> If you are running a version of android newer then 9 or 10 you can enable that feature this way usually:
> 
> Go to Settings > Network & internet.
> Select Advanced > Private DNS.
> Select the Private DNS provider hostname option.
> Enter one.one.one.one and select Save.

Does this seem to work for those affected?
